api/services/User/index.js

```diff
@@ -63,34 +63,22 @@ export default class User extends BaseModelService {
         ],
       });
     });
-    return Promise.all(resultPromise);
+    const result = await Promise.all(resultPromises);
+    return result.map((data) => !data);
   }
 
-  static availabilityLoader = new DataLoader(
-    this.batchFunctionForAvailabilityLoader,
-  );
-
-  static async findAvailaibilityDuring(
-    user_id,
-    dateTimeRange,
-    exculdeEvents = [],
-  ) {
-    const keyForDataLoader = {
-      user_id,
-      dateTimeRange,
-      exculdeEvents,
-    };
-    const data = await this.availabilityLoader.load(keyForDataLoader);
-    return !data;
+  static getAvailabilityLoader() {
+    return new DataLoader(this.findAllAvailabilityDuring);
   }
 
   @requireInstance
   ifAvailableDuring(dateTimeRange, exculdeEvents) {
-    return User.findAvailaibilityDuring(
-      this.instance.id,
+    const key = {
+      user_id: this.instance.id,
       dateTimeRange,
       exculdeEvents,
-    );
+    };
+    return User.getAvailabilityLoader().load(key);
```

Comment:
Here you don't need to integrate with loader. Just call `User.finAllAvailabilityDuring([key])`. Await and return the first item from the array as result. 

It can b done in 1 statement. No need to write 2 for that.

Reply:
```javascript
return (await User.finAllAvailabilityDuring([key]))[0];
```